src/datasources/accounts/accounts.datasource.spec.ts

```diff
@@ -184,6 +383,44 @@ describe('AccountsDatasource tests', () => {
       expect(actual).toStrictEqual(expect.arrayContaining(expected));
     });
 
+    it('should get the account data settings from cache', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const account = await target.createAccount(address);
+      const accountDataTypes = Array.from(
+        { length: faker.number.int({ min: 1, max: 4 }) },
+        () => accountDataTypeBuilder().with('is_active', true).build(),
+      );
+      const insertedDataTypes =
+        await sql`INSERT INTO account_data_types ${sql(accountDataTypes, 'name', 'is_active')} returning *`;
+      const accountDataSettingRows = insertedDataTypes.map((dataType) => ({
+        account_id: account.id,
+        account_data_type_id: dataType.id,
+        enabled: faker.datatype.boolean(),
+      }));
+      await sql`
+        INSERT INTO account_data_settings
+        ${sql(accountDataSettingRows, 'account_id', 'account_data_type_id', 'enabled')} returning *`;
+
+      // cache the account data settings
+      await target.getAccountDataSettings(address);
+
+      // delete the account data settings
+      await sql`DELETE FROM account_data_settings`;
+
+      // check the account data settings are still in the cache
+      const actual = await target.getAccountDataSettings(address);
```

Comment:
As before regarding database and cache out of sync.

Reply:
Answered in https://github.com/safe-global/safe-client-gateway/pull/1760#discussion_r1679010289